CRM/Event/Form/Search.php

```diff
@@ -284,7 +284,7 @@ private function submit($formValues) {
     }
 
     // We don't show test records in summaries or dashboards
-    if (empty($this->_formValues['participant_test']) && $this->_force) {
+    if (empty($this->_formValues['participant_test']) && $this->_force && !empty($this->_context) && $this->_context === 'dashboard') {
```

Comment:
There are a number of places that attempt to use `$this->_force` to decide whether test entities should be shown. I just did a test (without this PR) and the standard "Find participants" task still excludes test if you de-select the Yes/No on the first search.
I really don't want to see any changes to whether test entities are shown to rely on `$this->_force` because it is so inconsistent/unreliable. `$this->_force` is used to control whether the search results are shown immediately or whether you have to click "Search" first and should not be used for other things.

Reply:
I don't understand. The `_force` argument is not part of my change. It was already there. What I'm adding is the `_context` part. Do you mean that you want me to remove the `_force` part?

Reply:
@elcapo I know that it is not part of your change. But it is used to control display of test entities when it should not be. When we add more logic (eg. `_context`) to that we are making it even more difficult to fix later.
So yes I am proposing that any solution should remove the use of `_force` to control test entities.